internal/eval/eval.go

```diff
@@ -4816,165 +4816,43 @@ func evalZRANGE(args []string, store *dstore.Store) []byte {
 }
 
-// parseEncodingAndOffet function parses offset and encoding type for bitfield commands
-// as this part is common to all subcommands
-func parseEncodingAndOffset(args []string) (eType, eVal, offset interface{}, err error) {
-	encodingRaw := args[0]
-	offsetRaw := args[1]
-	switch encodingRaw[0] {
-	case 'i':
-		eType = SIGNED
-		eVal, err = strconv.ParseInt(encodingRaw[1:], 10, 64)
-		if err != nil {
-			err = diceerrors.NewErr(diceerrors.InvalidBitfieldType)
-			return eType, eVal, offset, err
-		}
-		if eVal.(int64) <= 0 || eVal.(int64) > 64 {
-			err = diceerrors.NewErr(diceerrors.InvalidBitfieldType)
-			return eType, eVal, offset, err
-		}
-	case 'u':
-		eType = UNSIGNED
-		eVal, err = strconv.ParseInt(encodingRaw[1:], 10, 64)
-		if err != nil {
-			err = diceerrors.NewErr(diceerrors.InvalidBitfieldType)
-			return eType, eVal, offset, err
-		}
-		if eVal.(int64) <= 0 || eVal.(int64) >= 64 {
-			err = diceerrors.NewErr(diceerrors.InvalidBitfieldType)
-			return eType, eVal, offset, err
-		}
-	default:
-		err = diceerrors.NewErr(diceerrors.InvalidBitfieldType)
-		return eType, eVal, offset, err
-	}
-
-	switch offsetRaw[0] {
-	case '#':
-		offset, err = strconv.ParseInt(offsetRaw[1:], 10, 64)
-		if err != nil {
-			err = diceerrors.NewErr(diceerrors.BitfieldOffsetErr)
-			return eType, eVal, offset, err
-		}
-		offset = offset.(int64) * eVal.(int64)
-	default:
-		offset, err = strconv.ParseInt(offsetRaw, 10, 64)
-		if err != nil {
-			err = diceerrors.NewErr(diceerrors.BitfieldOffsetErr)
-			return eType, eVal, offset, err
-		}
-	}
-	return eType, eVal, offset, err
-}
-
-// evalBITFIELD evaluates BITFIELD operations on a key store string, int or bytearray types
-// it returns an array of results depending on the subcommands
-// it allows mutation using SET and INCRBY commands
-// returns arity error, offset type error, overflow type error, encoding type error, integer error, syntax error
-// GET <encoding> <offset> -- Returns the specified bit field.
-// SET <encoding> <offset> <value> -- Set the specified bit field
-// and returns its old value.
-// INCRBY <encoding> <offset> <increment> -- Increments or decrements
-// (if a negative increment is given) the specified bit field and returns the new value.
-// There is another subcommand that only changes the behavior of successive
-// INCRBY and SET subcommands calls by setting the overflow behavior:
-// OVERFLOW [WRAP|SAT|FAIL]`
-func evalBITFIELD(args []string, store *dstore.Store) []byte {
-	if len(args) < 1 {
-		return diceerrors.NewErrArity("BITFIELD")
-	}
-
-	var overflowType string = WRAP // Default overflow type
-
-	type BitFieldOp struct {
-		Kind   string
-		EType  string
-		EVal   int64
-		Offset int64
-		Value  int64
-	}
-	var ops []BitFieldOp
-
-	for i := 1; i < len(args); {
-		switch strings.ToUpper(args[i]) {
+// This method executes each operation, contained in ops array, based on commands used.
+func executeOps(value *ByteArray, ops []utils.BitFieldOp) []interface{} {
```

Comment:
It's unclear what the executeOps method does by just looking at its naming.

```suggestion
func executeBitfieldOps(value *ByteArray, ops []utils.BitFieldOp) []interface{} {
```

Reply:
Updated to **executeBitfieldOps** in this [commit](https://github.com/DiceDB/dice/commit/b4601acfe13a6de265db09477dc84a3597e2fe26)